Geocoder/src/test/java/com/example/geocoder/steps/CensusApiServiceSteps.java

```diff
@@ -0,0 +1,81 @@
+package com.example.geocoder.steps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import com.example.geocoder.CucumberContextConfig;
+import com.example.geocoder.exceptions.InvalidAddressException;
+import com.example.geocoder.exceptions.MissingStreetException;
+import com.example.geocoder.exceptions.MissingZipAndCityStateException;
+import com.example.geocoder.requests.AddressRequest;
+import com.example.geocoder.responses.CensusApiResponse;
+import com.example.geocoder.services.CensusApiService;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+@SpringBootTest
+public class CensusApiServiceSteps extends CucumberContextConfig {
+
+    @Autowired
+    private CensusApiService censusApiService;
```

Comment:
Suggestion: BDD tests typically test the application as a whole and should be at least submitting the request to the controller.

Extra credit would be using `MockMvc` to send an http request instead of using the `Controller` directly.